src/code/z_demo.c

```diff
@@ -1232,683 +1401,778 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
                 break;
 
-            case 12:
-                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * temp);
+            case CS_TRANS_BLACK_HALF_FILL:
+                gSaveContext.cutsceneTransitionControl = 255.0f - (155.0f * lerp);
                 break;
 
-            case 13:
+            case CS_TRANS_BLACK_HALF_UNFILL:
                 play->envCtx.screenFillColor[0] = 0;
                 play->envCtx.screenFillColor[1] = 0;
                 play->envCtx.screenFillColor[2] = 0;
-                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - temp) * 155.0f);
+                play->envCtx.screenFillColor[3] = 255.0f - ((1.0f - lerp) * 155.0f);
                 break;
         }
     }
 }
 
-// Command 0x1 & 0x5: Camera Eye Points
-s32 Cutscene_Command_CameraEyePoints(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 relativeToLink) {
-    s32 shouldContinue = 1;
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_SetCamEyePoints(PlayState* play, CutsceneContext* csCtx, u8* scriptPtr, u8 relativeToPlayer) {
+    s32 shouldContinue = true;
+    CsCmdGeneric* cmd = (CsCmdGeneric*)scriptPtr;
     s32 size;
 
-    cmd += 8;
-    size = 8;
+    scriptPtr += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
+
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((csCtx->camEyePointsAppliedFrame < cmd->startFrame) || (csCtx->camEyePointsAppliedFrame >= 0xF000))) {
+        csCtx->camEyeReady = true;
+        csCtx->camEyePoints = (CutsceneCameraPoint*)scriptPtr;
+
+        if (csCtx->camAtReady) {
+            // the frame number set here isn't important, it just signals that the camera data has been applied
+            csCtx->camEyePointsAppliedFrame = cmd->startFrame;
 
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((csCtx->unk_18 < cmdBase->startFrame) || (csCtx->unk_18 >= 0xF000))) {
-        csCtx->unk_1B = 1;
-        csCtx->subCamEyePoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1A != 0) {
-            csCtx->unk_18 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+            if (gUseCutsceneCam) {
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_CS_0);
-                Play_ChangeCameraStatus(play, sReturnToCamId, CAM_STAT_WAIT);
+                Play_ChangeCameraStatus(play, sPrevCamId, CAM_STAT_WAIT);
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Camera_ResetAnim(Play_GetCamera(play, csCtx->subCamId));
-                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->subCamLookAtPoints,
-                                   csCtx->subCamEyePoints, GET_PLAYER(play), relativeToLink);
+                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->camLookAtPoints, csCtx->camEyePoints,
+                                   GET_PLAYER(play), relativeToPlayer);
```

Comment:
```suggestion
                Camera_SetCsParams(Play_GetCamera(play, csCtx->subCamId), csCtx->camLookAtPoints, csCtx->camEyePoints,
                                   GET_PLAYER(play), relativeToPlayer);
```
Or even `Camera_SetCutsceneParams`?

Reply:
i have no idea what happens in camera-land regarding this stuff. my knowledge ends here when z_demo passes it on. 
if you feel any of those names are right then sure, but maybe can be done with camera changes

Reply:
No worries, then just leave it as it.